### PR TITLE
fix: add migration for tslint to eslint

### PR DESCRIPTION
**Pull request by Jefiozie**
**Description:**
Migration from tslint to eslint.

This is the first step to migrate to eslint, I've not yet removed tslint but will do this after some feedback. I've used the typescript-eslint to preserve existing rules.

Looking forward on your feedback.

**Related issue (if exists):**
#5078 

p.s. probably some CI steps will fail with this change.

cc: @kwonoj @cartant

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,8 +27,10 @@
     "commitmsg": "validate-commit-msg",
     "build:spec:browser": "echo \"Browser test is not working currently\" && exit -1 && webpack --config spec/support/webpack.mocha.config.js",
     "decision_tree_widget": "cd doc/decision-tree-widget && npm run build && cd ../..",
-    "lint_spec": "tslint -c tslint.json \"spec/**/*.ts\"",
-    "lint_src": "tslint -c tslint.json \"src/**/*.ts\"",
+    "lint-spec": "tslint -c tslint.json \"spec/**/*.ts\"",
+    "lint_eslint_spec": "eslint --ext=ts spec/**/*  -c .eslintrc",
```
Comment (kwonoj) [inline package.json]: I'd like to suggest to follow `:` for script convention (using `-` was old way we did), like `lint:eslint:spec`.

btw, maybe with eslint we can just make single lit to run all spec / src at once? I think it was splitted as tslint glob behaves weird on windows.

File: tslint-to-eslint-config.log [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import-spacing does not yet have an ESLint equivalent.
```
Comment (kwonoj) [inline tslint-to-eslint-config.log]: do we need log file to be checked in?

File: package.json (hunk 1 above)
Comment (Jefiozie) [reply package.json]: Actually the tslint run scripts will be removed so I will do this in my next commit to give a good overview on what the change is.

File: tslint-to-eslint-config.log (hunk 2 above)
Comment (Jefiozie) [reply tslint-to-eslint-config.log]: Nope, good catch

File: package.json (hunk 1 above)
Comment (kwonoj) [reply package.json]: Comment was for new script `eslint`.

**Comment by Jefiozie** [comment]
So I made the changes that where mentioned in the comments. I have two questions remaining the migration.

1. Should we change the dtslint rule set also?
2. The lint rule fails atm however the settings are the same. Would you like me to fix them once by running the fix command?

**Comment by benlesh** [comment]
@kwonoj @cartant where are we at with this one? _I_ think it looks good, but I don't have a ton of experience with eslint.

**Comment by Jefiozie** [comment]
Hi @carant, any updates on this PR?

**Comment by cartant** [comment]
> Hi @cartant, any updates on this PR?

Sorry. I've not had time to look over it; I've had other priorities. Which is not to say that this isn't appreciated. It is and thanks for the time you've put into it. I'll bump it up a bit in my backlog and will try to make some time for it on the weekend.

**Comment by Jefiozie** [comment]
>Sorry. I've not had time to look over it; I've had other priorities. Which is not to say that this isn't appreciated. It is and thanks for the time you've put into it. I'll bump it up a bit in my backlog and will try to make some time for it on the weekend.

HI @cartant, thanks for your reply. Take your time I was just curious what the status was of this PR and If some of the team already had a look.

**Comment by Jefiozie** [comment]
I'm closing im going to close this one and look if I can retry it with the new changes.

**Pull request closed without merging**